### PR TITLE
[Inductor][CPP] Add Min/Max with VecMask

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #128825
* #128808
* #128804
* #128824
* #126504
* #126940
* __->__ #126841

**Summary**
Fix issue: https://github.com/pytorch/pytorch/issues/126824 which is missing the support of `min/max` with `VecMask`.

**TestPlan**
```
python test/inductor/test_torchinductor_opinfo.py -k test_comprehensive_clamp_max_cpu_bool
python test/inductor/test_torchinductor_opinfo.py -k test_comprehensive_clamp_min_cpu_bool
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by isuruf — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/codegen/cpp.py [hunk 1]
```diff
@@ -1308,11 +1315,21 @@ def truncdiv(a, b):
 
     @staticmethod
     def minimum(a, b):
-        return f"at::vec::minimum({a}, {b})"
+        if a.dtype == torch.bool:
+            assert b.dtype == torch.bool
+            a_cast, b_cast = unify_mask_base_type(V.kernel.compute, (a, b))
+            return f"decltype({a_cast})::minimum({a_cast}, {b_cast})"
```
Comment (peterbell10) [inline torch/_inductor/codegen/cpp.py:1321]: NIT: `min(a, b)` on boolean is just `a & b`.  Similarly `max(a, b)` is `a | b`.

**Review by peterbell10 — APPROVED** [review]

File: torch/_inductor/codegen/cpp.py (hunk 1 above)
Comment (leslie-fang-intel) [reply torch/_inductor/codegen/cpp.py:1321]: Thanks for the comment. Simplify to use `a & b` and `a | b`.

**Comment by leslie-fang-intel** [comment]
Hi @isuruf, these 2 PRs in the ghstack may have conflict with https://github.com/pytorch/pytorch/pull/124021. We prefer to land 124021 at first. Then I will rebase to land the PRs here. Does it sound good to you?

**Comment by isuruf** [comment]
Sounds good to me.

**Comment by DanilBaibak** [comment]
Hi @leslie-fang-intel! Sorry, I need to revert your PRs because they block reverting of broken PR. Please rebase your PRs and merge them as soon as CI is green.

File: test/inductor/test_torchinductor_opinfo.py [hunk 2]
```diff
@@ -413,8 +413,6 @@ def wrapper_noop_set_seed(op, *args, **kwargs):
     "addmv": {f16},
     "argsort": {b8, f16, f32, f64, i32, i64},
     "as_strided.partial_views": {f16},
-    "clamp_max": {b8},
-    "clamp_min": {b8},
```
Comment (isuruf) [inline test/inductor/test_torchinductor_opinfo.py:417]: Aside from these two, you can remove `max.binary, min.binary, maximum, minimum` too

Comment (leslie-fang-intel) [reply test/inductor/test_torchinductor_opinfo.py:417]: Oh, just saw the comment. I will send a standalone PR to remove it.

**Pull request closed without merging**